patoolib/programs/p7zip.py

```diff
@@ -101,51 +109,61 @@ def test_7z (archive, compression, cmd, verbosity, interactive):
   test_7z
 
 
-def create_7z(archive, compression, cmd, verbosity, interactive, filenames):
+def create_7z(archive, compression, cmd, verbosity, interactive, filenames, password=None):
     """Create a 7z archive."""
     cmdlist = [cmd, 'a']
     if not interactive:
         cmdlist.append('-y')
+    if password:
+        cmdlist.append('-p%s' % password)
```

Comment:
these two lines are common to many invocations. Having a helper

```
def _maybe_add_password(cmdline, password):
    if password:
        cmdlist.append('-p%s' % password)
```
and use it instead would allow to centralize the logic and thus in the future possibly add a single location to fix up/extend (if we discover that some characters aren't legit etc)

Reply:
similarly for other tools as well

Reply:
Agreed, it's better. As I understood you propose separate helper for every program. Maybe will be better to add some global helper for all programs? Something like this:
```
def _maybe_add_password(cmdlist, password, added_args):
    if password:
        if isinstance(added_args, list):
            cmdlist.extend(added_args)
        else:
            cmdlist.append(added_args)
```
and use it like:
```
from . import maybe_add_password
maybe_add_password(cmdlist, password, '-p%s' % password)
```
or 
```
maybe_add_password(cmdlist, password, ['-p', password])
```

Reply:
yeah, I was also considering generic helper, but decided against it -- it would indeed then duplicate all the specifications of password args for each invocation, thus not much really of help.  So probably sticking with a dedicated for each program would be of the most benefit

Reply:
Changed